src/deskolem.ml

```diff
@@ -0,0 +1,49 @@
+open Core
+open Terms
+open Timed
+
+(** [subst_inv t s x] replaces the term starting with the symbol [s] by a fresh
+    variable [x] in the term [t]. *)
+let rec subst_inv : term -> sym -> term Bindlib.var -> term =
+  fun t s x ->
+  let h, ts = Basics.get_args t in
+  match h with
+  | Vari _
+  | Type
+  | Kind                      -> t
+  | Symb(s', _) when s == s'  -> Vari x
```

Comment:
don't forget to add the arguments ts! you should instead have: `Basics.add_args (Vari x) ts`

Reply:
Why do I need to add `ts` here?
We want to replace the term: `s t1 t2 ... t3` with a fresh variable `x`. If `s` match our symbol then we replace all the term with `x`.

Reply:
Ah right. In this case, indeed, don't do `let ts = List.map subst ts` before the match.